### PR TITLE
[HTML] Add variable for custom tag start character

### PR DESCRIPTION
**Pull request by deathaxe**
This commit adds a `tag_name_start` variable to represent a custom tag's first character across syntax definitions instead of hard-coding `[A-Za-z]`.

It enables template syntaxes to override such tags' names easier, if they begin with a template interpolation.

Example:

    <{{tag_name}} ...>

To scope `{{tag_name}}` entity.name.tag, a template syntax has needed to extend the `tag-other` context before this commit with something like:

    tag-other:
      - meta_prepend: true
      - match: </?(?={{) 
        scope: punctuation.definition.tag.begin.html 
        push:
          - tag-other-content
          - tag-other-name

even though `{{ ... }}` tags are already part of the `prototype` context.

With this commit it is enough to override the `tag_name_start` to add the template begin punctuation in order to achieve the same outcome.

    variables:
      tag_name_start: (?:[A-Za-z]|{{)

> [!NOTE]
>
> The new variable is added to all template syntaxes such as ASP, ERB, JSP and PHP in order to enable support for tag names, beginning with template interpolation.

### CODE REVIEW

**Review by michaelblyons — APPROVED** [review]
I like this a lot.

File: HTML/HTML (Plain).sublime-syntax [hunk 1]
```diff
@@ -88,7 +88,8 @@ variables:
   tag_name_break_char: '[{{ascii_space}}/<>]'
   tag_name_break: (?={{tag_name_break_char}})
   tag_name_char: '[^{{tag_name_break_char}}]'
-  tag_name: '[A-Za-z]{{tag_name_char}}*'
+  tag_name_start: '[A-Za-z]'
```
Comment (FichteFoll) [inline HTML/HTML (Plain).sublime-syntax:91]: How about adding a comment here that explains the intent of this variable, i.e. being used for extensions? Will make it more obvious in the future that this is something external syntaxes (or internal even) actually depend on and reduces the likelyhood of changing/removing it by accident.

**Review by FichteFoll — DISMISSED** [review]

File: HTML/HTML (Plain).sublime-syntax (hunk 1 above)
Comment (deathaxe) [reply HTML/HTML (Plain).sublime-syntax]: Added a note and some docs at the beginning of the HTML (Plain) syntax.

**Review by michaelblyons — APPROVED** [review]

File: HTML/HTML (Plain).sublime-syntax [hunk 2]
```diff
@@ -60,6 +60,24 @@
 #          pop: 1
 #        - ...
 #
+#  2 TAG NAME INTERPOLATION
+#
+#    Default HTML tags start with a case insensitive latter a..z.
```
Comment (FichteFoll) [inline HTML/HTML (Plain).sublime-syntax:65]: `case-insensitive letter`

File: HTML/HTML (Plain).sublime-syntax [hunk 3]
```diff
@@ -60,6 +60,24 @@
...
+#
+#    If a template language supports interpolation within HTML tag names,
+#    it must also ensure a tag being identified as such, if its name starts
+#    with a template tag by overriding the `tag_name_start` variable.
+#
+#    Example:
+#
+#      variables:
+#        # make `{{` a valid tag name start char
```
Comment (FichteFoll) [inline HTML/HTML (Plain).sublime-syntax:74]: `start sequence`

**Review by FichteFoll — APPROVED** [review]

**Review by michaelblyons — APPROVED** [review]

**Pull request merged**